src/com/rakovets/course/challenge/oop/polimorphism/battle_ground/Archer.java

```diff
@@ -9,12 +9,19 @@ public Archer(String name, int health) {
     @Override
     public void takeDamage(int damageFromEnemy) {
         this.health -= damageFromEnemy;
-        System.out.printf("\nCurrent HP of hero: %d\n", this.health);
+        if (isAlive()) {
+            System.out.printf("\nCurrent HP of hero: %d\n", this.health);
+        } else {
+            System.out.println("\nArcher is dead");
+        }
+
```

Comment:
if implementations method for all subclasses are equal then move implementation to superclass